knit/__init__.py

```diff
@@ -5,10 +5,15 @@
 from .core import *
 from .env import CondaCreator, zip_path
 from .yarn_api import YARNAPI
-try:
-    from .dask_yarn import DaskYARNCluster
-except ImportError:    # pragma: no cover
-    warnings.warn('dask/distributed not installed, '
-                  'DaskYARNCluster not available')
+
+
+class DaskYARNCluster(object):
+    def __new__(cls, *args, **kwargs):
+        import dask_yarn
+        warnings.warn("Deprecated use of DaskYARNCluster in the knit "
+                      "namespace. Use the dask_yarn package instead.")
+        o = dask_yarn.DaskYARNCluster.__new__(dask_yarn.DaskYARNCluster)
+        o.__init__(*args, **kwargs)
```

Comment:
This has a few problems:
- Class is available even if `dask_yarn` isn't (e.g. dask isn't installed). This changes from the previous behavior where the class was only there if dask was installed.
- Incorrect use of `__new__`. You shouldn't call `__init__` in `__new__`, python will do it for you:

```python
In [1]: class Foo(object):
   ...:     def __init__(self, a, b):
   ...:         print("Foo.__init__(%s, %s)" % (a, b))
   ...:

In [2]: class Bar(Foo):
   ...:     def __new__(cls, a, b):
   ...:         print("Bar.__new__(%s, %s)" % (a, b))
   ...:         return super(Bar, cls).__new__(cls)
   ...:

In [3]: Bar(1, 2)
Bar.__new__(1, 2)
Foo.__init__(1, 2)
Out[3]: <__main__.Bar at 0x10ac47630>
```
- Checking subclasses doesn't work. `isinstance(DaskYARNCluster(), DaskYARNCluster)` is False.

Instead, I'd just subclass and override the `__init__`:

```python
try:
    from .dask_yarn import DaskYARNCluster as _DaskYARNCluster

    class DaskYARNCluster(_DaskYARNCluster):
        def __init__(self, *args, **kwargs):
            warnings.warn("DeprecationWarning: `knit.DaskYARNCluster` is deprecated, "
                          "please use `dask_yarn.DaskYARNCluster` instead")
            super(DaskYarnCluster, self).__init__(*args, **kwargs)
except ImportError:
    pass
```